src/datasources/email-template/pushwoosh-template.service.ts

```diff
@@ -1,24 +1,71 @@
 import { Injectable } from '@nestjs/common';
 import { getAddress } from 'viem';
 import { IEmailTemplate } from '@/domain/interfaces/email-template.interface';
+import { Chain } from '@/routes/chains/entities/chain.entity';
+import { readFileSync } from 'fs';
+import { join } from 'path';
 
 @Injectable()
 export class PushWooshTemplate implements IEmailTemplate {
-  addressToHtml(address: string): string {
+  addressToSafeWebAppUrl(args: { chain: Chain; safeAddress: string }): string {
+    return `https://app.safe.global/home?safe=${args.chain.shortName}:${args.safeAddress}`;
+  }
+
+  addressListToHtml(args: { chain: Chain; addresses: Array<string> }): string {
+    const listItems = args.addresses.map((owner) => {
+      const checksummedAddress = getAddress(owner);
+
+      const addressHtml = this.addressToHtml({
+        chain: args.chain,
+        address: checksummedAddress,
+      });
+      const explorerLinkHtml = this.addressToExplorerLink({
+        chain: args.chain,
+        address: checksummedAddress,
+      });
+
+      return `<li>${addressHtml}${explorerLinkHtml}</li>`;
+    });
+
+    return `<ul>${listItems.join('')}</ul>`;
+  }
+
+  private addressToHtml(args: { chain: Chain; address: string }): string {
     const CSS_ID = 'address-center';
 
-    const checksummedAddress = getAddress(address);
+    const checksummedAddress = getAddress(args.address);
 
     const start = checksummedAddress.slice(0, 6);
     const center = checksummedAddress.slice(6, -4);
     const end = checksummedAddress.slice(-4);
 
-    return `${start}<span id="${CSS_ID}">${center}</span>${end}`;
+    return `<span>${start}<span id="${CSS_ID}">${center}</span>${end}</span>`;
+  }
+
+  private addressToExplorerLink(args: {
+    chain: Chain;
+    address: string;
+  }): string {
+    const ADDRESS_TEMPLATE = '{{address}}';
```

Comment:
Nit: what about removing this variable and using the value string inline? It's only used once in the file. Alternatively, we could make it static if you prefer to keep the variable reference.

Reply:
This was moved in a40eb1ca59a440ffd68a3a4a1316dc59cc05fe0f.